File: java-parent/resources-core/src/main/java/de/cfranzen/archsonar/resources/filesystem/PathResource.java
package de.cfranzen.archsonar.resources.filesystem;

import de.cfranzen.archsonar.resources.Resource;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.apache.tika.mime.MediaType;

import java.io.IOException;
import java.io.InputStream;
import java.net.URI;
import java.nio.file.Files;
import java.nio.file.Path;

@EqualsAndHashCode(of = "path")
@ToString(of = "uri")
public class PathResource implements Resource {

    private final Path path;

    private final URI uri;

    PathResource(final Path path) {
        this.path = path;
        this.uri = path.toAbsolutePath().toUri();
    }

    @Override
    public URI uri() {
        return uri;
    }

    @Override
    public InputStream openInputStream() {
        try {
            return Files.newInputStream(path);
        } catch (IOException e) {
            throw new IllegalStateException("Unable to open InputStream for " + path, e);
        }
    }

    @Override
    public MediaType type() {
        return MediaType.OCTET_STREAM;
    }
}

File: kotlin-parent/resources-filesystem/src/main/kotlin/de/cfranzen/archsonar/resources/filesystem/PathResource.kt
package de.cfranzen.archsonar.resources.filesystem

import de.cfranzen.archsonar.resources.Resource
import java.io.InputStream
import java.net.URI
import java.nio.file.Files
import java.nio.file.Path

class PathResource(private val path: Path) : Resource {

    override val uri: URI = path.toAbsolutePath().toUri()

    override fun openInputStream(): InputStream {
        return Files.newInputStream(path)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as PathResource
        if (uri != other.uri) return false
        return true
    }

    override fun hashCode(): Int {
        return uri.hashCode()
    }

    override fun toString(): String {
        return "PathResource($uri)"
    }


}